tutorials/en/vectara-beginner-app-tutorial-showcase-the-creation-of-vectara-app-in-legal-use-case.mdx

```diff
@@ -0,0 +1,380 @@
+    finally:
+        os.remove(tmp_filename)
+```
+
+In this snippet, user input is captured through Streamlit's `st.chat_input`, Vectara is queried for relevant knowledge content, and Langchain processes the input to generate a response. Additionally, new documents are added to Vectara to enrich the knowledge base when a submit action occurs.
+
+---
+
+### Final Result: Grasping Your Law and Securing Legal Assistance
+
+
+Embarking on the venture of 'The Creation Of Vectara App In Legal Use Case' has been a remarkable journey. The culmination of this project has resulted in a platform that's not just technically sound but one that stands as a beacon of legal assistance for those in need. The Vectara App is designed to be a haven of legal knowledge, making legal understanding and help accessible to everyone.
+
+#### **1. A Visual Tour Through the Interface:**
+
+Our application exudes a user-centric design, ensuring that navigating through the multitude of legal information is a breeze. Each section is meticulously crafted to offer a seamless user experience.
+
+Upload your PDF and enter your keys and submit to proceed!
+
+<Img 
+  src="https://i.postimg.cc/4dSW3JDS/Screenshot-2023-11-02-at-2-42-36-PM.png" 
+  alt="Interface 1" 
+  href="https://postimg.cc/LYjtCM6t" 
+/>
+
+#### **2. Legal Knowledge Base: Your Personal Legal Advisor**
+
+The core of the Vectara App lies in its robust legal knowledge base. Users can dive into a sea of legal information, understand laws, and find answers to their legal queries with just a few clicks.
+
+<Img 
+  src="https://i.postimg.cc/dtbtyrRq/legal.png" 
+  alt="legal" 
+  href="https://postimg.cc/56mVcHkG" 
+/>
+
+---
+
+## Conclusion
+
+
+The expedition of creating the Vectara App in a legal use case scenario has not only been an avenue of technical exploration but a venture into making legal help more accessible. Through this project, a platform has been constructed where legal understanding is not confined to the experts, but is available to all.
+
+The Vectara App stands as a testament to the power of blending legal expertise with cutting-edge technology. As you navigate through the app, the ease with which you can now understand and seek legal help is apparent. This project is a stride towards demystifying the legal realm and making legal assistance a part of the daily lives of individuals.
+
+As we wrap up, the potential for further enhancement and the impact that the Vectara App can create in the legal domain is colossal. The horizon is vast and inviting, promising a realm where legal assistance is not a hurdle but a companion in everyone's life.
+
+---
+
+### **Live Demo and Further Exploration**
+
+Experience the application firsthand and delve deeper into its mechanics.For a deeper dive into the code and underlying mechanisms, visit the project on Hugging Face:
+
+
+**[Try it on Hugging Face](https://huggingface.co/spaces/Sanchayt/VectaraBeginner)**
```

Comment:
How can I try the final app?

Reply:
You can try it now @ofermend

Reply:
Great. works nice. Perhaps you can provide one or two legal PDF documents people can use as examples?

Reply:
Can do that , @ezzcodeezzlife can I add drive link for the pdfs or is there a specific format for pdf?

Reply:
@ofermend  I have incorporated the suggestions and pushed it , can you review this tutorial again?